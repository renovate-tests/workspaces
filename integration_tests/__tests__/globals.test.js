/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @flow
 */

'use strict';

const path = require('path');
const os = require('os');
const skipOnWindows = require('../../scripts/skip_on_windows');
const runJest = require('../runJest');
const {extractSummary} = require('../utils');
const {createEmptyPackage, writeFiles, cleanup} = require('../utils');

const DIR = path.resolve(os.tmpdir(), 'global-variables.test');
const TEST_DIR = path.resolve(DIR, '__tests__');

skipOnWindows.suite();

beforeEach(() => {
  cleanup(DIR);
  createEmptyPackage(DIR);
});

afterAll(() => cleanup(DIR));

test('basic test constructs', () => {
  const filename = 'basic.test-constructs.test.js';
  const content = `
    it('it', () => {});
    test('test', () => {});

    describe('describe', () => {
      it('it', () => {});
      test('test', () => {});
    });
  `;

  writeFiles(TEST_DIR, {[filename]: content});
  const {stderr, status} = runJest(DIR);
  expect(status).toBe(0);

  const {summary, rest} = extractSummary(stderr);
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
});

test('skips', () => {
  const filename = 'skips-constructs.test.js';
  const content = `
    it('it', () => {});
    xtest('xtest', () => {});
    xit('xit', () => {});
    it.skip('it.skip', () => {});
    test.skip('test.skip', () => {});

    xdescribe('xdescribe', () => {
      it('it', () => {});
      test('test', () => {});
    });

    describe.skip('describe.skip', () => {
      test('test', () => {});
      describe('describe', () => {
        test('test', () => {});
      });
    });
  `;

  writeFiles(TEST_DIR, {[filename]: content});
  const {stderr, status} = runJest(DIR);

  const {summary, rest} = extractSummary(stderr);
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
  expect(status).toBe(0);
});

test('only', () => {
  const filename = 'only-constructs.test.js';
  const content = `
    it('it', () => {});
    test.only('test.only', () => {});
    it.only('it.only', () => {});
    fit('fit', () => {});

    fdescribe('fdescribe', () => {
      it('it', () => {});
      test('test', () => {});
    });

    describe.only('describe.only', () => {
      test('test', () => {});
      describe('describe', () => {
        test('test', () => {});
      });
    });
  `;

  writeFiles(TEST_DIR, {[filename]: content});
  const {stderr, status} = runJest(DIR);
  expect(status).toBe(0);

  const {summary, rest} = extractSummary(stderr);
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
});

test('tests with no implementation', () => {
  const filename = 'only-constructs.test.js';
  const content = `
    it('it', () => {});
    it('it, no implementation');
    test('test, no implementation');
  `;

  writeFiles(TEST_DIR, {[filename]: content});
  const {stderr, status} = runJest(DIR);
  expect(status).toBe(0);

  const {summary, rest} = extractSummary(stderr);
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
});

test('skips with expand arg', () => {
  const filename = 'skips-constructs.test.js';
  const content = `
    it('it', () => {});
    xtest('xtest', () => {});
    xit('xit', () => {});
    it.skip('it.skip', () => {});
    test.skip('test.skip', () => {});

    xdescribe('xdescribe', () => {
      it('it', () => {});
      test('test', () => {});
    });

    describe.skip('describe.skip', () => {
      test('test', () => {});
      describe('describe', () => {
        test('test', () => {});
      });
    });
  `;

  writeFiles(TEST_DIR, {[filename]: content});
  const {stderr, status} = runJest(DIR, ['--expand']);
  expect(status).toBe(0);

  const {summary, rest} = extractSummary(stderr);
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
});

test('only with expand arg', () => {
  const filename = 'only-constructs.test.js';
  const content = `
    it('it', () => {});
    test.only('test.only', () => {});
    it.only('it.only', () => {});
    fit('fit', () => {});

    fdescribe('fdescribe', () => {
      it('it', () => {});
      test('test', () => {});
    });

    describe.only('describe.only', () => {
      test('test', () => {});
      describe('describe', () => {
        test('test', () => {});
      });
    });
  `;

  writeFiles(TEST_DIR, {[filename]: content});
  const {stderr, status} = runJest(DIR, ['--expand']);
  expect(status).toBe(0);

  const {summary, rest} = extractSummary(stderr);
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
});

test('tests with no implementation with expand arg', () => {
  const filename = 'only-constructs.test.js';
  const content = `
    it('it', () => {});
    it('it, no implementation');
    test('test, no implementation');
  `;

  writeFiles(TEST_DIR, {[filename]: content});
  const {stderr, status} = runJest(DIR, ['--expand']);
  expect(status).toBe(0);

  const {summary, rest} = extractSummary(stderr);
  expect(rest).toMatchSnapshot();
  expect(summary).toMatchSnapshot();
});
