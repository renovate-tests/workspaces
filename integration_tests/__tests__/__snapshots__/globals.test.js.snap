// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic test constructs 1`] = `
" PASS  __tests__/basic.test-constructs.test.js
  ✓ it
  ✓ test
  describe
    ✓ it
    ✓ test

"
`;

exports[`basic test constructs 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`only 1`] = `
" PASS  __tests__/only-constructs.test.js
  ✓ test.only
  ✓ it.only
  ✓ fit
  ○ skipped 1 test
  fdescribe
    ✓ it
    ✓ test
  describe.only
    ✓ test
    describe
      ✓ test

"
`;

exports[`only 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       1 skipped, 7 passed, 8 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`only with expand arg 1`] = `
" PASS  __tests__/only-constructs.test.js
  ○ it
  ✓ test.only
  ✓ it.only
  ✓ fit
  fdescribe
    ✓ it
    ✓ test
  describe.only
    ✓ test
    describe
      ✓ test

"
`;

exports[`only with expand arg 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       1 skipped, 7 passed, 8 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`skips 1`] = `
" PASS  __tests__/skips-constructs.test.js
  ✓ it
  ○ skipped 4 tests
  xdescribe
    ○ skipped 2 tests
  describe.skip
    ○ skipped 1 test
    describe
      ○ skipped 1 test

"
`;

exports[`skips 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       8 skipped, 1 passed, 9 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`skips with expand arg 1`] = `
" PASS  __tests__/skips-constructs.test.js
  ✓ it
  ○ xtest
  ○ xit
  ○ it.skip
  ○ test.skip
  xdescribe
    ○ it
    ○ test
  describe.skip
    ○ test
    describe
      ○ test

"
`;

exports[`skips with expand arg 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       8 skipped, 1 passed, 9 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`tests with no implementation 1`] = `
" PASS  __tests__/only-constructs.test.js
  ✓ it
  ○ skipped 2 tests

"
`;

exports[`tests with no implementation 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       2 skipped, 1 passed, 3 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`tests with no implementation with expand arg 1`] = `
" PASS  __tests__/only-constructs.test.js
  ✓ it
  ○ it, no implementation
  ○ test, no implementation

"
`;

exports[`tests with no implementation with expand arg 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       2 skipped, 1 passed, 3 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;
